server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -532,10 +531,50 @@ public void receive(Mutation m) {
     log.debug("TABLET_HIST {} opened", extent);
   }
 
+  private long setUpVolumeChooserInterval(TableConfiguration tblConf) {
+    long generalVolumeChooseSetting = tblConf
+        .getTimeInMillis(Property.VOLUME_CHOOSER_UPDATE_INTERVAL);
+    String tableSpecificSetting = tblConf.get(TABLE_VOLUME_CHOOSER_INTERVAL_PROPERTY);
+    long tableSpecificVolumeChooserSetting = tableSpecificSetting == null ? 0
+        : ConfigurationTypeHelper.getTimeInMillis(tableSpecificSetting);
+    if (tableSpecificVolumeChooserSetting > 0) {
+      return tableSpecificVolumeChooserSetting;
+    } else {
+      return  generalVolumeChooseSetting;
+    }
+  }
+
   public ServerContext getContext() {
     return context;
   }
 
+  private void updateTabletLocation() {
+    Path locationPath;
+    if (volumeChooserUpdateInterval > 0
+        && (lastVolumeChooserUpdate + volumeChooserUpdateInterval < System.currentTimeMillis())) {
+      VolumeManager fs = this.tabletServer.getFileSystem();
+      UniqueNameAllocator namer = context.getUniqueNameAllocator();
+      VolumeChooserEnvironment chooserEnv = new VolumeChooserEnvironment(extent.getTableId(),
+          context);
+
+      Text endRow = extent.getEndRow();
+      String directory = fs.choose(chooserEnv,
+          ServerConstants.getBaseUris(context.getConfiguration())) + Constants.HDFS_TABLES_DIR
```

Comment:
I'll make some time to review this closer, but first thought was how does this handle namespaces?